### ISSUE TITLE
panan-005 crash

### ISSUE THREAD

**Issue by adele157** [COSIMA/mom6-panan#34]
I'm getting a crash in the panan-005 case (test-1/20deg branch) in June of the 2nd year with the new bathymetry. 

`WARNING from PE     1: Extreme surface sfc_state detected: i=1711 j= 139 lon=-194.475 lat= -78.184 x=-194.475 y= -78.184 D= 1.0433E+01
 SSH=-1.7514E+00 SST=-2.4306E+00 SSS= 4.5011E+01 U-= 3.6473E-02 U+= 9.8926E-03 V-= 0.0000E+00 V+= 2.2825E-02`

`FATAL from PE    14: There were a total of         1 locations detected with extreme surface values!`

This occurred with both DT = 450 and DT = 360. The old panan-005 was stable with DT = 360. I can try to rerun from one month earlier with DT = 360, but also wanted to flag this here, because it would be good to plot the bathymetry in this location to see what it looks like. Maybe a bathymetry edit is needed?

**Comment by adele157** [comment]
This crash wasn't fixed by rerunning with DT = 360 from the previous month. I can try reducing the timestep further to get through this, but we know that the old 1/20th panant-v2 was stable at this point with DT = 360. Any chance someone could plot up the bathymetry in this region (`i=1711 j= 139, lon=-194.475 lat= -78.184`) from this file: `/g/data/e14/cs6673/panan-005/domain-tools/new_topog/topog.nc` ?

@schmidt-christina, @fabiobdias, @willaguiar?

**Comment by willaguiar** [comment]
> This crash wasn't fixed by rerunning with DT = 360 from the previous month. I can try reducing the timestep further to get through this, but we know that the old 1/20th panant-v2 was stable at this point with DT = 360. Any chance someone could plot up the bathymetry in this region (`i=1711 j= 139, lon=-194.475 lat= -78.184`) from this file: `/g/data/e14/cs6673/panan-005/domain-tools/new_topog/topog.nc` ?
> 
> @schmidt-christina, @fabiobdias, @willaguiar?

hi @adele157 I don't have access to e14, [but I'm assuming](https://github.com/willaguiar/miscellanea/blob/19749cff52fdc6b2a387c9d259ab77cc4f1331a0/Bathymetry.ipynb) this is the same file as /scratch/v45/cs6673/shared/edit_topog_new.nc (@schmidt-christina?)
It seems to be a place along the Ross Sea Shelf. At that specific I,j point depth=12.263889m

<img width="576" alt="Screen Shot 2023-03-29 at 9 46 46 AM" src="https://user-images.githubusercontent.com/70033934/228384519-7f747b12-e2be-48db-81e1-cb3354a42a41.png">

<img width="572" alt="Screen Shot 2023-03-29 at 10 13 02 AM" src="https://user-images.githubusercontent.com/70033934/228387401-d1d65949-609f-49af-b016-d2fbca40e635.png">

**Comment by fabiobdias** [comment]
Yeah it has something weird near the crash location. I just looked at the bathymetry Adele's pointed and there are a few grid points quite deep (> 1700m) close to the coast in the western Ross Sea.

![mom6-panan-006_topography_RossSea_crash](https://user-images.githubusercontent.com/8631257/228388045-adc20108-6580-4062-bd7f-ae2654d1980e.png)

**Comment by willaguiar** [comment]
Why are these files different? I thought we were using the edited bathymetry.

**Comment by fabiobdias** [comment]
I just plotted a comparison between the 2 files, it seems the exact crash location (in between the 4 red dots) is southwest of this deep feature (which exists in both topography files)
![mom6-panan-005_topography_RossSea_crash](https://user-images.githubusercontent.com/8631257/228390388-3b82d635-0945-43d8-9fda-ee74f5ff05fb.png)

**Comment by adele157** [comment]
Hmmm doesn't look like there's any weird embayments or anything. But that whole south-west corner is really shallow and the cells there will be really small horizontally also. The model just crashed again at the same time even with DT = 300. 

What are people's thoughts on what we should do here? 

What about deepening the bathymetry in that whole region (south of 77.8S or so) to a minimum of 50m? Is that too extreme?

@fabiobdias could you do a zoom in plot of south of 77S in that region with an adjusted colorbar?

**Comment by fabiobdias** [comment]
![Screenshot 2023-03-29 at 10 50 09 am](https://user-images.githubusercontent.com/8631257/228391734-660064d2-72b5-42cb-b191-66cea19310c1.png)

**Comment by aekiss** [comment]
Could you also you plot the old and new topog in that region with the same colour scale, and the difference between them?

**Comment by fabiobdias** [comment]
![Screenshot 2023-03-29 at 10 56 32 am](https://user-images.githubusercontent.com/8631257/228392413-71b3529b-35a7-438b-a021-46faba854693.png)

**Comment by AndyHoggANU** [comment]
I would ask here, what is extreme about the surface state? It seems that SSH is -1.75m, which is low, but maybe realistic? SSS is -2.43, also very low, and I am not certain how this can happen with our EOS? The default `BAD_VAL_SST_MIN` is -2.1. What are you using?

Another options is to set `CHECK_BAD_SURFACE_VALS` to `False` and see if it keeps running?

**Comment by adele157** [comment]
@fabiobdias is this comparing with the old 1/20th, i.e. panant-v2, this topography file: `/g/data/x77/ahg157/inputs/mom6/panan/ocean_topog_filled.nc` ?

**Comment by angus-g** [comment]
> I would ask here, what is extreme about the surface state?

I think you misread the SST and SSS: SST is -2.43, pretty low, and SSS is 45!

**Comment by aekiss** [comment]
I guess the extreme value it's complaining about is `SSS= 4.5011E+01`?

**Comment by aekiss** [comment]
what does the sea ice look like?

**Comment by adele157** [comment]
Loving all the help guys. :smiley:

This is the last day of the month before it crashes. Thickness looks ok? Concentration is close to 1 there and there are also some weird features in the sea ice further north? Maybe that's convection? Looks like the high concentration of sea ice in that little south-west corner might be stuck there?

<img width="1167" alt="Screen Shot 2023-03-29 at 11 14 34 am" src="https://user-images.githubusercontent.com/8506963/228395030-21fb345e-f503-4190-8d46-40350aa430a2.png">

<img width="1167" alt="Screen Shot 2023-03-29 at 11 17 47 am" src="https://user-images.githubusercontent.com/8506963/228395067-a0d1368d-dac9-45a8-93c7-bb474c4eb1a2.png">

**Comment by aekiss** [comment]
SIC looks numerically unstable (checkerboarding)
![Screen Shot 2023-03-29 at Wed 29-3 11 31am](https://user-images.githubusercontent.com/31054815/228396401-a9f7ef4b-48a3-4b21-b6d1-a38091f0d327.png)

**Comment by adele157** [comment]
Not sure, this is shortest timestep I've run (DT = 300). I just checked the
old 1/20th panan (panant-v2) and it has the same checkerboarding.

**Comment by aekiss** [comment]
OK so both configs have problems in the sea ice. Even if that's not relevant to this crash, it probably wouldn't do good things for brine rejection or surface fluxes.

**Comment by aekiss** [comment]
I've pulled this out as a separate issue

**Comment by adele157** [comment]
Also crashes with `DT = 225`. Noone else is keen on the bathymetry editing idea?

**Comment by aekiss** [comment]
Topo is the same in [this comparison](https://github.com/COSIMA/mom6-panan/issues/34#issuecomment-1487755267), so what changes would you suggest?

**Comment by adele157** [comment]
Yeah, though in that earlier panant-v2 simulation there were bugs in the sea ice, e.g. This shows average Antarctic sea ice thickness in the original 1/20th panan simulations (blue and orange) we're comparing against):

<img width="1189" alt="Screen Shot 2023-03-30 at 12 55 12 pm" src="https://user-images.githubusercontent.com/8506963/228708126-cf43b850-79cf-4133-8eac-7f53a88d927f.png">

So it's not really a clean comparison to a working simulation.

I suggest either:
a) Deepening the bathymetry in that whole region south of ~78S, or
b) Turning it into land.

@fabiobdias was going to make some plots comparing the 1/20th bathymetry to the original GEBCO, which might help us decide.

**Comment by aekiss** [comment]
Might also be interesting to compare to the 0.1deg topo

**Comment by fabiobdias** [comment]
yeah I'm working on it right now and can include the 0.1deg topo. One thing I can confirm is that the ~1700m deep hole seen [here](https://github.com/COSIMA/mom6-panan/issues/34#issuecomment-1487728116) at approximately (-198W, -78S) exists in the GEBCO too. 

Anyone have the full path of the 0.1deg topo?

**Comment by adele157** [comment]
panan-01 bathymetry is here:
`/g/data/ik11/inputs/mom6/panan/01deg/20230314/ocean_topog.nc`

**Comment by fabiobdias** [comment]
Here we go:

![GEBCO2022xMOM6-panan-005_topography_map](https://user-images.githubusercontent.com/8631257/228731077-eff84ff4-88d2-4719-bbdb-6c40540e0360.png)

**Comment by AndyHoggANU** [comment]
OK, in that case, I think I am sold on Adele's land masking suggestion.

**Comment by adele157** [comment]
Fast ice mapping from [Fraser et al. 2020](https://essd.copernicus.org/articles/12/2987/2020/) (orange is always fast ice, yellow is sometimes fast ice):

<img width="400" alt="Screen Shot 2023-03-30 at 12 55 12 pm" src="https://user-images.githubusercontent.com/8506963/228732393-6238368f-da62-435d-92f5-a44478bdfd98.png">

So it's an area of permanent fast ice. Let's kill it and turn it into land. Maybe anything less than 100m deep?

@fabiobdias and @schmidt-christina, do you reckon you could work on this?

**Comment by adele157** [comment]
Let's do the same in the 1/40th bathymetry @willaguiar.

**Comment by fabiobdias** [comment]
yeah happy to give it a go

**Comment by aekiss** [comment]
Sounds good to me - the model can't represent fast ice so may be getting blown offshore and generating too much sea ice there ([it's very thin, despite the high SIC](https://github.com/COSIMA/mom6-panan/issues/34#issuecomment-1487773967), so is likely to be very young ice), hence the excessive SSS. 

We can play fast and loose with the "coastline" here, as it's actually shelf/fast ice and subject to change anyway...

**Comment by fabiobdias** [comment]
Do you have a topography editing tool to suggest @aekiss ?

**Comment by adele157** [comment]
I have some new values for the sea ice time steps I'd like to try in the 1/20th running from scratch again. It would be good to incorporate this bathymetry edit also, so we can run out for a few more years of testing to identify any new issues before we start in earnest on the 1/40th run. Has anyone started on editing the bathymetry yet? @fabiobdias @schmidt-christina? If not, I can just do a quick edit now, just let me know.

**Comment by adele157** [comment]
I edited the bathymetry, new file is here:
`/g/data/v45/akm157/inputs/mom6/panan/panan_005/topog_edited.nc`. 

I don't know the workflow for making masks and have run out of time now. If anyone else has a chance to make new masks for this new bathymetry let me know and I can set it running again with new topo and new sea ice time steps.

**Comment by aekiss** [comment]
probably too late now, but re.
> Do you have a topography editing tool to suggest @aekiss ?

for hand-edits of a few points, you could try https://github.com/COSIMA/topogtools/blob/master/editTopo.py

**Comment by adele157** [comment]
Update on this: the Ross Sea bathymetry edit worked.
